openapi3.yaml

```diff
@@ -1,57 +1,143 @@
+  /getProducts:
     get:
-      operationId: getAnotherResource
+      operationId: getProducts
       tags:
-        - anotherResource
-      summary: gets the resource
+        - getProducts
+      summary: get products from db catalog, accroding to the filter in requestBody
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/requestBodyForFilter'          
       responses:
         200:
-          description: OK
+          description: A list of products
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/anotherResource'
-  /resourceName:
-    get:
-      operationId: getResourceName
+                type: array
+                items:
+                  $ref: '#/components/schemas/product'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        404:
+          description: NOT FOUND
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        500:
+          description: SERVER ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+
+  /addProduct:
+    post:
+      operationId: addProduct
       tags:
-        - resourceName
-      summary: gets the resource
+        - addProduct
+      summary: creates a new record of type product
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/product'          
+      responses:
+        201:
+          description: created, send back the product that added
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/product'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        500:
+          description: SERVER ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+
+  /updateProduct/{id}:
+    patch:
+      operationId: updateProduct
+      tags:
+        - updateProduct
+      summary: update the chosen feilds of a product
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/product'      
       responses:
         200:
-          description: OK
+          description: updated, send back the product that updated
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/resource'
-    post:
-      operationId: createResource
+                $ref: '#/components/schemas/product'
```

Comment:
When `patching` a product it is returned in the response? Why?

Reply:
just to let the user make sure everything worked as expected.